### ISSUE TITLE
directAccess + query.aggregate + dates no longer work

### ISSUE THREAD

**Issue by jonas-db** [parse-community/parse-server#7748]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description

When you start the server with the option `directAccess`, the execution of a pipeline does not return results when you pass a `Date` instance to `match`.

### Steps to reproduce

1. The following test _fails_ when `directAccess` is set to `true` and `'date': date`
2. The following test _passes_ when `directAccess` is set to `false` and `'date': date`
3. The following test _passes_ when `directAccess` is set to `true` or `false` and `date : date.toISOString()`.

It remains unclear whether this behaviour was intended in previous versions (<5)

```
    it('fails', async () => {
        const registration = await Parse.User.signUp("jonas", "test", {});
        expect(registration.id).to.be.exist

        const user = await Parse.User.logIn("jonas", "test");
        expect(user.get('sessionToken')).to.be.exist
        expect(user.id).to.be.exist

        const TestObject = Parse.Object.extend('TestObject');

        const date = new Date()
        const testObject = await new TestObject({
            date: date,
        }).save({}, { useMasterKey : true });

        const match = {
            'date': { "$gte": date, "$lte": date } // Does not work when directAccess is true
            //'date': { "$gte": date.toISOString(), "$lte": date.toISOString()} // Works for both
        }
        
        console.log(match)

        const pipeline = [
            {
                "$match" : match
            }
        ]
        const query = new Parse.Query(TestObject);
        const result = await query.aggregate(pipeline).catch(e => e);
        console.log("results=",util.inspect(result, { showHidden: false, depth: null }))

        const query2 = new Parse.Query(TestObject);
        const result2 = await query2.equalTo('date', date).find().catch(e => e);
        console.log("results2=",util.inspect(result2, { showHidden: false, depth: null }))

        expect(result).to.be.lengthOf(1)
        expect(result2).to.be.lengthOf(1)
        expect(result[0].objectId).to.be.equal(result2[0].id)
    });
```

### Actual Outcome

When (1)
```
results= []
results2= [
  ParseObjectSubclass {
    className: 'TestObject',
    _objCount: 9,
    id: 'jXEY7xFC3a'
  }
]
```

### Expected Outcome

When (2) or (3)
```
results= [
  {
    date: { __type: 'Date', iso: '2021-12-28T13:51:11.961Z' },
    createdAt: '2021-12-28T13:51:11.962Z',
    updatedAt: '2021-12-28T13:51:11.962Z',
    objectId: 'xNLwllPXlu'
  }
]
results2= [
  ParseObjectSubclass {
    className: 'TestObject',
    _objCount: 9,
    id: 'xNLwllPXlu'
  }
]
```

### Environment

Server
- Parse Server version: `^5.0.0-beta.4`
- Operating system: `macos`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `local`

Database
- System (MongoDB or Postgres): `mongodb`
- Database version: `5.0.3`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `local`

Client
- SDK (iOS, Android, JavaScript, PHP, Unity, etc): `na`
- SDK version: `na`

**Comment by mtrezza** [comment]
Could you open a PR with a simplified version of the test so we can reproduce this? Simplified meaning that it should only contain lines that are required to reproduce the issue. Instead of logging in a user you can run the query with masterkey.

Are you observing the issue also on 4.x or only 5.x versions?

**Comment by mtrezza** [comment]
@jonas-db Do you have any update on this issue?

**Comment by dblythy** [comment]
@jonas-db I have attempted to replicate this and have not been successful. I wrote this failing test:

```js
it('fails', async () => {
    await reconfigureServer({directAccess: true});
    const TestObject = Parse.Object.extend('TestObject');
    const date = new Date()
    await new TestObject({ date: date }).save(null, { useMasterKey : true });

    const match = {
        'date': { "$lte": new Date()}
    }

    console.log(match)

    const pipeline = [
        {
            "$match" : match
        }
    ]
    const query = new Parse.Query(TestObject);
    const result = await query.aggregate(pipeline).catch(e => e);
    console.log("results=", result);

    const query2 = new Parse.Query(TestObject);
    const result2 = await query2.equalTo('date', date).find().catch(e => e);
    console.log("results2=", result2);

    expect(result.length).toBe(1);
    expect(result2.length).toBe(1);
    expect(result[0].objectId).toEqual(result2[0].id)
```

This passes on Parse Server.

I also attempted to run this by calling this function on a real Parse Server with `directAccess: true`:

(async () => {
  Parse.serverURL = // redacted
  const TestObject = Parse.Object.extend('TestObject');
  const date = new Date()
  await new TestObject({ date: date }).save(null, { useMasterKey : true });
  console.log('b')

  const match = {
      'date': { "$lte": new Date()}
  }

  console.log(match)

  const pipeline = [
      {
          "$match" : match
      }
  ]
  const query = new Parse.Query(TestObject);
  const result = await query.aggregate(pipeline).catch(e => e);
  console.log("results=", result);
})();


In both cases, the aggregate returns one object as a result.

Could you please clarify the what the failing test should be.

**Comment by dblythy** [comment]
I was able to confirm this issue on a real Parse Server. The issue is that `directAccess` passes a `Date` object directly to the pipeline, and the server thinks it should be a string (as normally JSON.parse() can't convert dates) and attempts to convert the string to a date. The request payload is the same in both cases, however for `directAccess`, the pipeline object is an `instanceof Date`, whereas without it is a string. It results in the pipeline being converted to `{}`, resulting in no matches.

The fix is very simple.

I attempted to write a failing test for this on Parse Server using `directAccess` but was unable, as the pipeline date was always stringified in the test suite.

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.3.0-alpha.24](https://github.com/parse-community/parse-server/releases/tag/5.3.0-alpha.24)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.4.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/5.4.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.4.0-alpha.1](https://github.com/parse-community/parse-server/releases/tag/5.4.0-alpha.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.4.0](https://github.com/parse-community/parse-server/releases/tag/5.4.0)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.4.0](https://github.com/parse-community/parse-server/releases/tag/5.4.0)
